### PR TITLE
internal(envelope): Convert SentrySdkInfo to Swift

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :scroll: Description

Converting SentrySdkInfo to Swift is a first change in order to add missing packages, needed for hybrid SDKs SR.

No functionality changed.

## :green_heart: How did you test it?
tests

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] I updated the wizard if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

## :crystal_ball: Next steps
- https://github.com/getsentry/sentry-cocoa/pull/4624

### CODE REVIEW

**Comment by brustolin** [comment]
Im curious about the motivation behind this change.

File: Sources/Swift/Protocol/SentrySdkInfo.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    static let PACKAGE_MANAGER: SentrySdkPackageManager = .spm
+#elseif COCOAPODS
+    static let PACKAGE_MANAGER: SentrySdkPackageManager = .cocoapods
+#elseif CARTHAGE_YES
+    // CARTHAGE is a xcodebuild build setting with value `YES`, we need to convert it into a compiler
+    // definition to be able to use it.
+    static let PACKAGE_MANAGER: SentrySdkPackageManager = .carthage
+#else
+    static let PACKAGE_MANAGER: SentrySdkPackageManager = .unknown
+#endif
```
Comment (krystofwoldrich) [inline Sources/Swift/Protocol/SentrySdkInfo.swift:26]: @brustolin @philipphofmann What is the easiest way to check that these compiler directives work also in Swift?

Comment (krystofwoldrich) [reply Sources/Swift/Protocol/SentrySdkInfo.swift:26]: I see for example SWIFT_PACKAGE we already use in swift so that should work.

**Comment by krystofwoldrich** [comment]
Seems like we won't be able to use Swift implementation at the moment, as it introduces issues with order of entries in serialized SdkInfo.

Example: 

```
{"event_id":"cdfae94e74a84efb87360dddc5bc0abb","sdk":{"name":"sentry.cocoa","version":"8.42.0-beta.2"}}
{"type":"event","length":88}
{"event_id":"cdfae94e74a84efb87360dddc5bc0abb","platform":"cocoa","timestamp":978307200}
```

is not equal to 

```
{"event_id":"cdfae94e74a84efb87360dddc5bc0abb","sdk":{"version":"8.42.0-beta.2","name":"sentry.cocoa"}}
{"type":"event","length":88}
{"event_id":"cdfae94e74a84efb87360dddc5bc0abb","platform":"cocoa","timestamp":978307200}
```

**Pull request closed without merging**